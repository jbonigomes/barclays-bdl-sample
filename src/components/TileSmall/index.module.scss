.container {
  width: 212px;
  font-size: 16px;
  line-height: 24px;
  text-align: center;
  font-family: 'ExpertSans-Light';

  img {
    width: 50px;
    margin-bottom: 12px;
  }

  .content {
    padding: 24px;
  }
}
