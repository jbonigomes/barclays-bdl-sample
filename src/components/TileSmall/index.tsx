import React from 'react'
import Tile from '../Tile'
import LeadLink from '../LeadLink'
import styles from './index.module.scss'

type TileSmallProps = {
  title: string,
  image: string,
  headerLevel?: number,
  children: React.ReactNode,
  onClick(e: React.MouseEvent<HTMLElement>): void,
}

const TileSmall: React.SFC<TileSmallProps> = ({
  title,
  image,
  onClick,
  children,
  headerLevel,
}: TileSmallProps) => (
  <div className={styles.container}>
    <Tile>
      <div className={styles.content}>
        <img src={image} alt="Tile graphic" />
        <LeadLink
          lightHover
          text={title}
          onClick={onClick}
          headerLevel={headerLevel}
        />
        {children}
      </div>
    </Tile>
  </div>
)

export default TileSmall
