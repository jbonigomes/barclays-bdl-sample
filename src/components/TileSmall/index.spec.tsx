import React from 'react'
import renderer from 'react-test-renderer'

import TileSmall from '.'

test('Match snapshot', () => {
  const props = {
    title: 'title',
    image: 'image',
    onClick: jest.fn(),
  }

  const tree = renderer.create(
    <TileSmall title={props.title} image={props.image} onClick={props.onClick}>
      Foo
    </TileSmall>
  )

  expect(tree.toJSON()).toMatchSnapshot()
})
