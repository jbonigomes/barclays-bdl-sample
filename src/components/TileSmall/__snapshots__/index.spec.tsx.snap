// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshot 1`] = `
<div
  className="container"
>
  <div
    className="tile"
  >
    <div
      className="content"
    >
      <img
        alt="Tile graphic"
        src="image"
      />
      <h1>
        <button
          className="lead lightHover"
          onClick={[MockFunction]}
        >
          title
        </button>
      </h1>
      Foo
    </div>
  </div>
</div>
`;
